### PR TITLE
refactor: use @ts-check wherever applicable

### PR DESCRIPTION
**Pull request by devanshbatra04**
So I applied ```@ts-check``` on all files involving `@type` occurences in ```src/```
This gives a bunch of errors, the import package errors should be fixed by #1988 but there are also other strange ones like 
```error TS2339: Property 'querySelectorAll' does not exist on type 'Node'```
and 
``` error TS2339: Property 'title' does not exist on type 'Element' ```

### CODE REVIEW

**Comment by devanshbatra04** [comment]
I think I should manually define the types where it fails. Analogous to https://stackoverflow.com/questions/31237237/property-queryselector-does-not-exist-on-type-node-in-typescript

File: src/core/dfn.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // @ts-check
 // Module core/dfn
 // - Finds all <dfn> elements and populates conf.definitionMap to identify them.
+// @ts-check
```
Comment (saschanaz) [inline src/core/dfn.js]: This is redundant 🤣

**Comment by saschanaz** [comment]
We should cast the types where appropriate rather than manually adding types.

File: src/core/dfn.js (hunk 1 above)
Comment (marcoscaceres) [inline src/core/dfn.js]:
```suggestion
```

**Comment by marcoscaceres** [comment]
As @saschanaz says, you will probably need to cast for:

> Property 'querySelectorAll' does not exist on type 'Node'

Which should be cast to `Element`. 

Likewise, `HTMLElement` has the title attribute:
https://html.spec.whatwg.org/multipage/dom.html#htmlelement

File: src/core/dfn.js (hunk 1 above)
Comment (devanshbatra04) [reply src/core/dfn.js]: LOL, my bad :sweat_smile:

**Comment by marcoscaceres** [comment]
Just want to say we appreciate the work you did here, @devanshbatra04. It motivated us to merge in there other PR, giving us really good TypeScript coverage.

**Comment by devanshbatra04** [comment]
Yep, I will work on the other files! Didn't know PRs close that way.

**Pull request closed without merging**